openpype/plugins/publish/extract_slate_global.py

```diff
@@ -0,0 +1,538 @@
+import pyblish.api
+from html2image import Html2Image
+
+from openpype.lib import (
+    get_oiio_tools_path,
+    get_ffprobe_streams,
+    get_chrome_tool_path,
+    run_subprocess,
+)
+from openpype.pipeline import publish
+
+
+def timecode_to_frames(timecode, framerate):
+    rt = otio.opentime.from_timecode(timecode, framerate)
+    return int(otio.opentime.to_frames(rt))
+
+
+def frames_to_timecode(frames, framerate):
+    rt = otio.opentime.from_frames(frames, framerate)
+    return otio.opentime.to_timecode(rt)
+
+
+def frames_to_seconds(frames, framerate):
+    rt = otio.opentime.from_frames(frames, framerate)
+    return otio.opentime.to_seconds(rt)
+
+
+class SlateCreator:
+    """Class that formats and renders an html template to use as a slate.
+
+    Html renders as sRGB (or Rec.709 if you want a little gamma boost), so
+    rendering will happen in .png as intermediate file, while then converted
+    to dest cspace using oiio --colorconvert attributes syntax.
+
+    """
+
+    def __init__(
+        self,
+        template_path,
+        resources_path="",
+        staging_dir="",
+        staging_subfolder="",
+        data=None,
+    ):
+        self.data = data.copy() if data else {}
+        self.log = logging.getLogger(self.__class__.__name__)
+
+
+        self.set_template_paths(template_path, resources_path=resources_path)
+        self.set_staging_dir(staging_dir, subfolder=staging_subfolder)
+        self.read_template()
+        self.task_filter = []
+
+    def set_template_paths(self, template_path, resources_path=""):
+        """Set path of HTML slate template file and CSS resources."""
+
+        self.template_path = os.path.normpath(template_path)
+        self.log.debug(
+            "Using Slate template path: '%s'", self.template_path
+        )
+
+        if resources_path:
+            self.resources_path = os.path.normpath(resources_path)
+            self.log.debug(
+                "Using Slate resources path: '%s'", self.resources_path
+            )
+        else:
+            basedir = os.path.basename(self.template_path)
+            self.resources_path = os.path.normpath(basedir)
+            self.log.debug(
+                "No Slate resources path specified, using: '%s'", self.resources_path
+            )
+
+    def set_resolution(self, width, height):
+        self.data["resolution_width"] = width
+        self.data["resolution_height"] = height
+        self.log.debug("Slate resolution set to: '%sx%s'", width, height)
+
+    def set_staging_dir(self, path="", subfolder=""):
+        if path:
+            staging_dir = os.path.normpath(path)
+        else:
+            staging_dir = tempfile.gettempdir()
+
+        if subfolder:
+            staging_dir = os.path.join(staging_dir, subfolder)
+
+        if not os.path.exists(staging_dir):
+            os.makedirs(staging_dir)
+
+        self.staging_dir = staging_dir
+        self.log.debug("Staging dir set to: '%s'", staging_dir)
```

Comment:
Is this confusing maybe that this uses a different staging dir data than the one from the publishing instance?